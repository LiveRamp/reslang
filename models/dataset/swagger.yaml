# generated by Reslang v0.9.0
openapi: 3.0.1
info:
    title: Segment API - ALPHA
    description: ""
    version: 0.0.1
servers:
    - url: "https://api.liveramp.com/segment-api"
tags:
    - name: Segment
      description: (asset) A canonical Segment
    - name: SegmentIngestionRequest
      description: "(request) "
    - name: SegmentExtractionRequest
      description: "(request) "
paths:
    /v1/segments:
        post:
            tags:
                - Segment
            operationId: Create Segment
            description: ""
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: "#/components/schemas/SegmentInput"
            responses:
                "201":
                    description: Segment created successfully
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    id:
                                        description: ""
                                        type: string
    "/v1/segments/{id}":
        get:
            tags:
                - Segment
            operationId: Get 1 Segment
            description: ""
            responses:
                "200":
                    description: Segment retrieved successfully
                    content:
                        application/json:
                            schema:
                                $ref: "#/components/schemas/SegmentOutput"
            parameters:
                - in: path
                  name: id
                  required: true
                  description: ""
                  schema:
                      type: string
    /v1/segment-ingestion-requests:
        post:
            tags:
                - SegmentIngestionRequest
            operationId: Create SegmentIngestionRequest
            description: ""
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: "#/components/schemas/SegmentIngestionRequestInput"
            responses:
                "201":
                    description: SegmentIngestionRequest created successfully
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    id:
                                        description: ""
                                        type: string
    "/v1/segment-ingestion-requests/{id}":
        get:
            tags:
                - SegmentIngestionRequest
            operationId: Get 1 SegmentIngestionRequest
            description: ""
            responses:
                "200":
                    description: SegmentIngestionRequest retrieved successfully
                    content:
                        application/json:
                            schema:
                                $ref: "#/components/schemas/SegmentIngestionRequestOutput"
            parameters:
                - in: path
                  name: id
                  required: true
                  description: ""
                  schema:
                      type: string
    /v1/segment-extraction-requests:
        post:
            tags:
                - SegmentExtractionRequest
            operationId: Create SegmentExtractionRequest
            description: ""
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: "#/components/schemas/SegmentExtractionRequestInput"
            responses:
                "201":
                    description: SegmentExtractionRequest created successfully
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    id:
                                        description: ""
                                        type: string
    "/v1/segment-extraction-requests/{id}":
        get:
            tags:
                - SegmentExtractionRequest
            operationId: Get 1 SegmentExtractionRequest
            description: ""
            responses:
                "200":
                    description: SegmentExtractionRequest retrieved successfully
                    content:
                        application/json:
                            schema:
                                $ref: "#/components/schemas/SegmentExtractionRequestOutput"
            parameters:
                - in: path
                  name: id
                  required: true
                  description: ""
                  schema:
                      type: string
components:
    parameters: {}
    schemas:
        SegmentInput:
            type: object
            properties:
                name:
                    description: ""
                    type: string
                identifierType:
                    description: ""
                    type: string
                    enum: &ref_0
                        - PEL
                        - CID
                        - MUID
                        - LRCookie
                        - PartnerCookie
            required:
                - name
                - identifierType
        SegmentOutput:
            type: object
            properties:
                id:
                    description: ""
                    type: string
                name:
                    description: ""
                    type: string
                identifierType:
                    description: ""
                    type: string
                    enum: *ref_0
                createdAt:
                    description: ""
                    type: string
                    format: ISO8601 UTC date-time
                    example: "2019-04-13T03:35:34Z"
                updatedAt:
                    description: ""
                    type: string
                    format: ISO8601 UTC date-time
                    example: "2019-04-13T03:35:34Z"
            required:
                - id
                - name
                - identifierType
                - createdAt
                - updatedAt
        SegmentIngestionRequestInput:
            type: object
            properties:
                segmentId:
                    description: ""
                    type: string
                    example: Link to a Segment  resource via its id
                inputDataURL:
                    description: ""
                    type: string
                    format: url
                    example: "https://www.domain.com (url)"
            required:
                - segmentId
                - inputDataURL
        SegmentIngestionRequestOutput:
            type: object
            properties:
                id:
                    description: ""
                    type: string
                segmentId:
                    description: ""
                    type: string
                    example: Link to a Segment  resource via its id
                inputDataURL:
                    description: ""
                    type: string
                    format: url
                    example: "https://www.domain.com (url)"
                createdAt:
                    description: ""
                    type: string
                    format: ISO8601 UTC date-time
                    example: "2019-04-13T03:35:34Z"
                status:
                    description: ""
                    type: string
                    enum: &ref_1
                        - QUEUED
                        - IN_PROGRESS
                        - COMPLETED
                        - FAILED
                        - CANCELLED
                completedAt:
                    description: ""
                    type: string
                    format: ISO8601 UTC date-time
                    example: "2019-04-13T03:35:34Z"
                MetaData:
                    description: ""
                    $ref: "#/components/schemas/SIROutputMetadata"
            required:
                - id
                - segmentId
                - inputDataURL
                - createdAt
                - status
        SegmentExtractionRequestInput:
            type: object
            properties:
                segmentId:
                    description: ""
                    type: string
                    example: Link to a Segment  resource via its id
            required:
                - segmentId
        SegmentExtractionRequestOutput:
            type: object
            properties:
                id:
                    description: ""
                    type: string
                segmentId:
                    description: ""
                    type: string
                    example: Link to a Segment  resource via its id
                createdAt:
                    description: ""
                    type: string
                    format: ISO8601 UTC date-time
                    example: "2019-04-13T03:35:34Z"
                status:
                    description: ""
                    type: string
                    enum: *ref_1
                completedAt:
                    description: ""
                    type: string
                    format: ISO8601 UTC date-time
                    example: "2019-04-13T03:35:34Z"
                MetaData:
                    description: ""
                    $ref: "#/components/schemas/SEROutputMetadata"
            required:
                - id
                - segmentId
                - createdAt
                - status
        SIROutputMetadata:
            type: object
            properties:
                version:
                    description: ""
                    type: string
                totalInputRecordCount:
                    description: ""
                    type: number
            required:
                - version
                - totalInputRecordCount
        SEROutputMetadata:
            type: object
            properties:
                outputDataURL:
                    description: ""
                    type: string
                    format: url
                    example: "https://www.domain.com (url)"
                totalOutputRecordCount:
                    description: ""
                    type: number
            required:
                - outputDataURL
                - totalOutputRecordCount
        StandardError:
            type: object
            properties:
                httpStatus:
                    description: HTTP error status code for this problem
                    type: number
                errorCode:
                    description: "Service specific error code, more granular"
                    type: string
                message:
                    description: "General, human readable error message"
                    type: string
            required:
                - httpStatus
                - errorCode
                - message
