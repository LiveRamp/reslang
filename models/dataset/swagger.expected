# generated by Reslang v0.9.9
openapi: 3.0.1
info:
  title: Segment API - ALPHA
  description: ''
  version: 0.0.1
servers:
  - url: 'https://api.liveramp.com/dataset'
tags:
  - name: 'Dataset::Field'
    description: '(subresource)  '
paths:
  '/v1/datasets/{datasetId}/fields':
    post:
      tags:
        - 'Dataset::Field'
      operationId: 'Create Dataset::Field'
      description: Create a new Field
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/DatasetFieldInput'
      responses:
        '201':
          description: Field created successfully
          content:
            application/json:
              schema:
                type: object
                properties:
                  id:
                    description: ''
                    type: integer
                    format: int64
        '303':
          description: Field already exists
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Redirection'
      parameters:
        - in: path
          name: datasetId
          description: Id of parent Dataset
          required: true
          schema:
            type: integer
            format: int32
    get:
      tags:
        - 'Dataset::Field'
      operationId: 'List Dataset::Fields'
      description: Find existing Fields
      responses:
        '200':
          description: Fields retrieved successfully
          headers:
            X-Total-Count:
              description: Total number of Fields returned by the query
              schema:
                type: integer
                format: int32
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/DatasetFieldMultiResponse'
      parameters:
        - in: path
          name: datasetId
          description: Id of parent Dataset
          required: true
          schema:
            type: integer
            format: int32
        - in: query
          name: offset
          description: Offset of the Fields (starting from 0) to include in the response.
          schema:
            type: integer
            format: int32
            default: 0
        - in: query
          name: limit
          description: >-
            Number of Fields to return. If unspecified, 10 max will be returned.
            Maximum value for limit can be 100
          schema:
            type: integer
            format: int32
            default: 10
            maximum: 100
        - in: query
          name: id
          description: ''
          required: false
          schema:
            type: integer
            format: int64
        - in: query
          name: enabled
          description: ''
          required: false
          schema:
            type: boolean
  '/v1/datasets/{datasetId}/fields/{id}':
    get:
      tags:
        - 'Dataset::Field'
      operationId: 'Get 1 Dataset::Field'
      description: Get a Field
      responses:
        '200':
          description: Field retrieved successfully
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/DatasetFieldOutput'
        '404':
          description: Field not found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/StandardError'
      parameters:
        - in: path
          name: datasetId
          description: Id of parent Dataset
          required: true
          schema:
            type: integer
            format: int32
        - in: path
          name: id
          required: true
          description: ''
          schema:
            type: integer
            format: int64
    put:
      tags:
        - 'Dataset::Field'
      operationId: 'Modify a Dataset::Field'
      description: Add values to a Field
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/DatasetFieldPuttable'
      responses:
        '200':
          description: Field modified successfully
        '400':
          description: Values do not match Field type
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/StandardError'
        '403':
          description: Field is not enumerable
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/StandardError'
        '404':
          description: Field not found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/StandardError'
      parameters:
        - in: path
          name: datasetId
          description: Id of parent Dataset
          required: true
          schema:
            type: integer
            format: int32
        - in: path
          name: id
          required: true
          description: ''
          schema:
            type: integer
            format: int64
    delete:
      tags:
        - 'Dataset::Field'
      operationId: 'Delete a Dataset::Field'
      description: Disable a Field
      responses:
        '200':
          description: Field deleted successfully
        '404':
          description: Field not found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/StandardError'
      parameters:
        - in: path
          name: datasetId
          description: Id of parent Dataset
          required: true
          schema:
            type: integer
            format: int32
        - in: path
          name: id
          required: true
          description: ''
          schema:
            type: integer
            format: int64
components:
  parameters: {}
  schemas:
    DatasetFieldInput:
      type: object
      properties:
        name:
          description: ''
          type: string
        type:
          description: ''
          $ref: '#/components/schemas/FieldTypeEnum'
        enumerable:
          description: ''
          type: boolean
        valueCollectionType:
          description: ''
          $ref: '#/components/schemas/ValueCollectionTypeEnum'
        values:
          description: ''
          items:
            type: string
          type: array
      required:
        - name
        - type
        - enumerable
        - valueCollectionType
    DatasetFieldOutput:
      type: object
      properties:
        id:
          description: ''
          type: integer
          format: int64
        name:
          description: ''
          type: string
        type:
          description: ''
          $ref: '#/components/schemas/FieldTypeEnum'
        enumerable:
          description: ''
          type: boolean
        valueCollectionType:
          description: ''
          $ref: '#/components/schemas/ValueCollectionTypeEnum'
        values:
          description: ''
          items:
            type: string
          type: array
        enabled:
          description: ''
          type: boolean
        createdAt:
          description: ''
          type: string
          format: ISO8601 UTC date-time
          example: '2019-04-13T03:35:34Z'
        updatedAt:
          description: ''
          type: string
          format: ISO8601 UTC date-time
          example: '2019-04-13T03:35:34Z'
      required:
        - id
        - name
        - type
        - enumerable
        - valueCollectionType
        - createdAt
        - updatedAt
    DatasetFieldPuttable:
      type: object
      properties:
        values:
          description: ''
          items:
            type: string
          type: array
    DatasetFieldMultiResponse:
      type: object
      properties:
        fields:
          description: 'Array of retrieved Dataset::Fields'
          type: array
          items:
            $ref: '#/components/schemas/DatasetFieldOutput'
    FieldTypeEnum:
      type: string
      enum:
        - STRING
        - NUMERIC
    ValueCollectionTypeEnum:
      type: string
      enum:
        - SINGLETON
        - SET
        - MULTISET
    Redirection:
      type: object
      properties:
        existingFieldUrl:
          description: ''
          type: string
      required:
        - existingFieldUrl
    StandardError:
      type: object
      properties:
        httpStatus:
          description: HTTP error status code for this problem
          type: integer
          format: int32
        errorCode:
          description: 'Service specific error code, more granular'
          type: string
        message:
          description: 'General, human readable error message'
          type: string
      required:
        - httpStatus
        - errorCode
        - message

